### ISSUE TITLE
Another bug in strcmp

### ISSUE THREAD

**Issue by Daniel-Cortez** [Open-GTO/sa-mp-fixes#66]
There's a bug in `strcmp` whence it returns 0 if either of compared strings is empty, which is already fixed in fixes.inc.
But it turns out this function also has another bug:
```Pawn
#include <a_samp>

main()
{
    new result;

    static const str1[] = "a", str2[] = "abc";
    result = strcmp(str1, str2, false, 4);
    printf("strcmp(\"%s\", \"%s\", false, 4):\t\t%d", str1, str2, result);

    static const str3[] = "abcdefgh", str4[] = "abc";
    result = strcmp(str3, str4, false, 4);
    printf("strcmp(\"%s\", \"%s\", false, 4):\t%d", str3, str4, result);

    static const str5[] = "a", str6[] = "abcdef";
    result = strcmp(str5, str6, false, 4);
    printf("strcmp(\"%s\", \"%s\", false, 4):\t%d", str5, str6, result);
}
```
Output:
```
strcmp("a", "abc", false, 4):           -2
strcmp("abcdefgh", "abc", false, 4):    5
strcmp("a", "abcdef", false, 4):        -5
```
So what happens here is that `strcmp` is supposed to return 1, 0 or -1, but returns the difference in number of characters instead. It happens if the length of either of compared strings is less than the value of the 4'th argument (the number of characters to compare; defaults to `cellmax`) and one string matches another one until the first encountered EOS.

**Comment by Y-Less** [comment]
That's not a bug - that's exactly what strcmp is supposed to do.  It compares strings and returns the difference.  I'm still not even convinced that returning 0 with an empty string is a bug and not just behaviour other than what people expect since most people just think it is an `IsSame` function.  This, despite the fact that apparently I wrote the original "fix".

**Comment by Daniel-Cortez** [comment]
Well, that's pretty interesting since it was officially considered a bug and fixed in Pawn 4.0: <a href="https://github.com/compuphase/pawn/issues/13">https://github.com/compuphase/pawn/issues/13</a>
And even the documentation for the amxString module (String_Manipulation.pdf; appeared some time between 3.2 and 4.0) says that strcmp can only return 1, 0 or -1.

**Comment by Y-Less** [comment]
OK then.
